pyterrier/pipelines.py

```diff
@@ -365,6 +388,14 @@ def _apply_round(measure, value):
     elif len(names) != len(retr_systems):
         raise ValueError("names should be the same length as retr_systems")
 
+    # validate save_dir and resulting filenames
+    if save_dir is not None:
+        if not os.path.exists(save_dir):
```

Comment:
Any reason we can't make the path for them if it doesn't exist? I guess there's a risk it could create tons of subdirectories.

Should also check `isdir` instead, otherwise they could get an error later in the off-chance it's a file or something else.

Reply:
> Any reason we can't make the path for them if it doesn't exist? I guess there's a risk it could create tons of subdirectories.

Hmmm. Not sure. Lets leave as is for now?

> Should also check `isdir` instead, otherwise they could get an error later in the off-chance it's a file or something else.
I added this.